src/posix/platform/udp.cpp

```diff
@@ -71,7 +71,8 @@ static int FdFromHandle(void *aHandle)
 
 static bool IsLinkLocal(const struct in6_addr &aAddress)
 {
-    return aAddress.s6_addr[0] == 0xfe && aAddress.s6_addr[1] == 0x80;
+    return (aAddress.s6_addr[0] == 0xfe && aAddress.s6_addr[1] == 0x80) ||
+           (aAddress.s6_addr[0] == 0xff && aAddress.s6_addr[1] == 0x02);
```

Comment:
In OT's existing implementation, `IsLinkLocal` is used to check whether a unicast address is link local and there would be a separate `IsLinkLocalMulticast` to check whether a multicast address is link local. 

https://github.com/openthread/openthread/blob/master/src/core/net/ip6_address.cpp#L65
https://github.com/openthread/openthread/blob/master/src/core/net/ip6_address.cpp#L75

Better to keep consistent for such interfaces?